### PR TITLE
chore(rewrite-existing): rewrite existing release note

### PR DESCRIPTION
**Pull request by anis-campos**
adding a script to easily rewrite a existing release

### CODE REVIEW

File: .github/workflows/validate_pr.yml [hunk 1]
```diff
@@ -23,12 +23,18 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+      - run: |
+          # create the master branch
+          git branch ${{ github.base_ref }} origin/${{ github.base_ref }}
+          # create the pr branch
+          git branch ${{ github.head_ref }} origin/${{ github.head_ref }}
```
Comment (noirbee) [inline .github/workflows/validate_pr.yml:30]: Why do you need this after using `fetch-depth: 0` ?

File: action.yml [hunk 2]
```diff
@@ -23,6 +23,8 @@ inputs:
       This can be used to keep only relevant commits by filtering on relevant files.
       Example:
         path_filters: "src/ doc/ tests/*/qa/*"
+  target:
+    description: A rev1..rev2 string to be used to generate the commit list
```
Comment (noirbee) [inline action.yml:27]: Prefer `--from-ref` / `--to-ref`, à la `pre-commit`

File: changelog_generator/generator.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    fixes = CommitTree(
+        commit_type=":bug: Fixes", commits=get_commit_from_type(commits, "fix")
+    )
+    if fixes.commits:
+        trees.append(fixes)
+    reverts = CommitTree(
+        commit_type=":scream: Revert", commits=get_commit_from_type(commits, "revert")
+    )
+    if reverts.commits:
+        trees.append(reverts)
```
Comment (noirbee) [inline changelog_generator/generator.py]: Do I really need to read the five exact same `if`s ?

File: changelog_generator/generator.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    target: Optional[str] = None,
+    prefix: Optional[str] = None,
+    filter_paths: Optional[str] = None,
+):
+    repository = RepositoryManager(
+        uri=repository_path, prefix=prefix, filter_paths=filter_paths
+    )
+    if target:
+        commits = repository.from_target(target)
+        (previous_tag, current_tag) = target.split("..")
```
Comment (noirbee) [inline changelog_generator/generator.py]:
```suggestion
        previous_tag, current_tag = target.split("..")
```

File: changelog_generator/generator.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    else:
+        commits = repository.commits_since_last_tag
+        (previous_tag, current_tag) = (repository.previous_tag, repository.current_tag)
```
Comment (noirbee) [inline changelog_generator/generator.py]:
```suggestion
        previous_tag, current_tag = repository.previous_tag, repository.current_tag
```

File: changelog_generator/repository_manager.py [hunk 6]
```diff
@@ -60,14 +60,17 @@ def commits_since_last_tag(self) -> Sequence[Commit]:
             revision = f"{self.previous_tag}..{self.current_tag}"
         else:
             revision = self.current_tag
+        return self._get_commits(revision)
 
+    def _get_commits(self, revision: str) -> Sequence[Commit]:
```
Comment (noirbee) [inline changelog_generator/repository_manager.py:65]:
```suggestion
    def _get_commits(self, revision: str) -> tuple[Commit]:
```

File: changelog_generator/repository_manager.py [hunk 7]
```diff
@@ -60,14 +60,17 @@ def commits_since_last_tag(self) -> Sequence[Commit]:
...
             options["paths"] = self.filter_paths
-
         commits = self.repository.iter_commits(revision, **options)
-
         return tuple(
             Commit(hexsha=commit.hexsha, summary=commit.summary, message=commit.message)
             for commit in commits
         )
+
+    def from_target(self, target: str) -> Sequence[Commit]:
```
Comment (noirbee) [inline changelog_generator/repository_manager.py:75]:
```suggestion
    def from_target(self, target: str) -> tuple[Commit]:
```

File: rewrite_existing.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        help="A rev1..rev2 string to be used to generate the commit list",
+    )
+    parser.add_argument(
+        "filter_paths",
+        nargs="*",
+        help="A space separated list of path to be used to the commits that edited files within "
+        "them",
+    )
+    args = parser.parse_args()
+    (tag_n, tag_n1) = args.target.split("..")
```
Comment (noirbee) [inline rewrite_existing.py]:
```suggestion
    tag_n, tag_n1 = args.target.split("..")
```

**Review by noirbee — CHANGES_REQUESTED** [review]

File: .github/workflows/validate_pr.yml (hunk 1 above)
Comment (anis-campos) [reply .github/workflows/validate_pr.yml:30]: fetch does not create the branches, and to no messup the links generated, I can't just use `origin/master`

File: changelog_generator/generator.py (hunk 3 above)
Comment (anis-campos) [reply changelog_generator/generator.py]: right, but not my code 😢

File: changelog_generator/repository_manager.py (hunk 6 above)
Comment (anis-campos) [reply changelog_generator/repository_manager.py:65]: this repo is still in python 3.7. I could use `Tuple` I guess

File: .github/workflows/validate_pr.yml (hunk 1 above)
Comment (noirbee) [reply .github/workflows/validate_pr.yml:30]: You don't need the branches, you just need the commits they're referring to be available when running `git log` etc.

File: changelog_generator/repository_manager.py (hunk 6 above)
Comment (noirbee) [reply changelog_generator/repository_manager.py:65]: Didn't realized that, ignore it then

File: .github/workflows/validate_pr.yml (hunk 1 above)
Comment (anis-campos) [reply .github/workflows/validate_pr.yml:30]: yes but this needs branches to make a proper link here:
https://github.com/lumapps/changelog-generator/blob/rewrite_release_note/changelog_generator/changelog_template.jinja#L1-L2

File: action.yml (hunk 2 above)
Comment (anis-campos) [reply action.yml:27]: I prefer the mono argument to no to have to handle the fact that it only works when both are provided

**Review by tvuillemin — APPROVED** [review]

**Review by noirbee — APPROVED** [review]

**Pull request merged**